mllib/src/main/scala/org/apache/spark/ml/regression/LinearRegression.scala

```diff
@@ -209,7 +219,13 @@ class LinearRegression(override val uid: String)
 
     if (handlePersistence) instances.unpersist()
 
-    copyValues(new LinearRegressionModel(uid, weights, intercept))
+    val model = copyValues(new LinearRegressionModel(uid, weights, intercept))
```

Comment:
No need to call copyValues since this is in train(), not fit().  (train handles that and setParent)  Could you also please correct the use of copyValues earlier in this method?